src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp

```diff
@@ -230,6 +239,49 @@ void FwAutotuneAttitudeControl::checkFilters()
 	}
 }
 
+bool FwAutotuneAttitudeControl::isAuxEnableSwitchEnabled()
+{
+	manual_control_setpoint_s manual_control_setpoint{};
+	_manual_control_setpoint_sub.copy(&manual_control_setpoint);
+
+	float aux_enable_channel = NAN;
+
+	switch (_param_fw_at_man_aux.get()) {
+	case 0:
+		break;
+
+	case 1:
+		aux_enable_channel = manual_control_setpoint.aux1;
+		break;
+
+	case 2:
+		aux_enable_channel = manual_control_setpoint.aux2;
+		break;
+
+	case 3:
+		aux_enable_channel = manual_control_setpoint.aux3;
+		break;
+
+	case 4:
+		aux_enable_channel = manual_control_setpoint.aux4;
+		break;
+
+	case 5:
+		aux_enable_channel = manual_control_setpoint.aux5;
+		break;
+
+	case 6:
+		aux_enable_channel = manual_control_setpoint.aux6;
+		break;
+
+	default:
+		aux_enable_channel = NAN;
+		break;
+	}
+
+	return (aux_enable_channel != NAN) && (aux_enable_channel > .5f);
```

Comment:
The direct comparison of this float to NaN causes a compile time error in the sitl build. Thoughts on how best to refactor to resolve? @dagar

Reply:
You can use PX4_ISFINITE()